lib/puppet/type.rb

```diff
@@ -84,6 +85,10 @@ class Type
   # Comparing type instances.
   include Comparable
 
+  @types = {}
+  # Lock for managing types
+  @lock = Puppet::Concurrent::Lock.new
```

Comment:
This is nuts. Would it at least make sense to move this closer to where we include MetatypeManager, and maybe comment `@types`?

Reply:
Yeah that makes sense at least. I hate this a lot.